sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1205,6 +1205,70 @@ object functions {
   def pow(l: Double, rightName: String): Column = pow(l, Column(rightName))
 
   /**
+   * Returns the positive value of l mod r.
+   *
+   * @group math_funcs
+   * @since 1.5.0
+   */
+  def pmod(l: Column, r: Column): Column = Pmod(l.expr, r.expr)
+
+  /**
+   * Returns the positive value of l mod r.
+   *
+   * @group math_funcs
+   * @since 1.5.0
+   */
+  def pmod(l: Column, rightName: String): Column = pmod(l, Column(rightName))
+
+  /**
+   * Returns the positive value of l mod r.
+   *
+   * @group math_funcs
+   * @since 1.5.0
+   */
+  def pmod(leftName: String, r: Column): Column = pmod(Column(leftName), r)
```

Comment:
Let's keep just 2 APIs.
```scala
def pmod(l: Column, r: Column): Column
def pmod(leftName: String, rightName: String): Column
```
BTW, probably we need to give more meaningful names for the parameters.